### PR TITLE
Examples and readme

### PR DESCRIPTION
**Pull request by yjchoi1**
added examples and readme

### CODE REVIEW

**Review by kks32 — CHANGES_REQUESTED** [review]
Please don't include the whole GNS code here. It should be using the exisitng GNS. If you want to tie to a specific version of GNS code, then use gitsubmodule for that commit.

**Comment by yjchoi1** [comment]
Added the submodule with the specific commit.

**Review by kks32 — CHANGES_REQUESTED** [review]
Don't you need to checkout a specific commit of GNS? I thought it only worked with a specific version of GNS?

File: inverse_barrier/inverse.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    #         current_particle_type,
+    #         mpm_inputs,
+    #         timestep_stride=10,
+    #         write_path=f"{output_dir}/trj-{iteration}.gif")
+
+
+
+
+
+
```
Comment (kks32) [inline inverse_barrier/inverse.py]:
```suggestion
```

Remove these

File: inverse_barrier/optimizations.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+#         loss
+#
+#     """
+#     if penalty_scale is None:
+#         loss = torch.mean((runout_end_pred - runout_end_true) ** 2)
+#     else:
+#         penalty = penalty_scale *
+#         loss = torch.mean((runout_end_pred - runout_end_true) ** 2)
+#
+#     return loss
```
Comment (kks32) [inline inverse_barrier/optimizations.py]: If not used remove these?

File: inverse_velocity/loss_landscape.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import time
+import sys
+import os
+import numpy as np
+import json
+import pickle
+import torch.utils.checkpoint
+
+sys.path.append("/work2/08264/baagee/frontera/gns-main/")
```
Comment (kks32) [inline inverse_velocity/loss_landscape.py]: Don't use hardcoded paths especially on public repos. Others should be able to reproduce.

File: inverse_velocity/loss_landscape.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+from example.inverse_velocity.forward import rollout_with_checkpointing
+
+from gns import reading_utils
+from gns import data_loader
+from gns import train
+
+
+# Work dir
+path = "data/"
```
Comment (kks32) [inline inverse_velocity/loss_landscape.py]: If you are expecting it in `data/` either make sure the script does this or specify in the README

**Review by kks32 — CHANGES_REQUESTED** [review]

**Comment by yjchoi1** [comment]
I have included the requests.

**Pull request merged**